### ISSUE TITLE
Byzcoin Benchmarking

### ISSUE THREAD

**Issue by Raycoms** [dedis/cothority#2441]
Hi, I was looking through the repo and I was wondering if there is a "simple" way to startup 400 instances and run a benchmark with fixed sized blocks.

**Comment by ineiti** [comment]
There is a simulation framework where you can do this. Look for example at https://github.com/dedis/cothority/tree/master/byzcoin/simulation - which should have a README :(

In that directory, you can run the simulation with

```bash
go run . coins.toml
```

This sets up a 5-node system and runs a number of coin-transactions on them, then it waits for all transactions to be accepted.

You can find more information about running the simulations here:

https://github.com/dedis/onet/tree/master/simul

**Comment by Raycoms** [comment]
I now had the time to look deeper into this. Not sure I understood this right, but it seems I can either run this on one of the cloud platforms (Deterlab etc) which need a special signup or solely on localhost using this.

Is there an easy way I can get each process to run within a docker container on a distributed set of processes like Grid5000?

**Comment by ineiti** [comment]
We use mininet to add network bandwidth and latency control. Since a month ago, you can also run it in a virtual machine - check out the end of this:

https://github.com/dedis/onet/blob/master/simul/platform/MININET.md

**Comment by ineiti** [comment]
Yes, we thought of doing AWS deployments, too, but currently this is very low on our priority list, sorry :(

Using the virtual machine and Mininet, you can simulate up to 200 nodes on a powerful machine, with bandwidth and latency restriction between each pair of nodes. For more you either need to give up the network restrictions and use localhost, or you need to set up manually multiple servers and then use mininet on them.

**Comment by ineiti** [comment]
Theoretically it should be enough to copy https://github.com/dedis/onet/blob/master/simul/platform/mininet.go into a `docker-swarm.go` and implement the public methods to fit your deployment. But the devil will be in the details, most annoyingly the way the monitor works...

**Comment by Raycoms** [comment]
I'm very confused about running the simulation (on mini-net or localhost). The localhost description is basically non existent and mininet tells me to run: "go build && ./simul -platform mininet simul.toml" without saying from which folder to run it, and what simul.toml should contain.

I guessed that simul.toml is supposed to contain the simulation settings like:

```
Simulation = "ExampleHandlers"
Servers = 16
BF = 2
Rounds = 10
#SingleHost = true

Hosts
3
7
15
31
```

but there is still no executable simul file

root@nova-16:~/onet# go build && ./simul -platform mininet simul.toml
-bash: ./simul: Is a directory

**Comment by ineiti** [comment]
This should work:

```
git clone https://github.com/dedis/cothority
cd cothoriyt/byzcoin/simulation
go build && ./simulation -platform localhost coins.toml
```
